core/imageroot/usr/local/agent/pypkg/cluster/modules.py

```diff
@@ -274,9 +274,25 @@ def list_available(rdb, skip_core_modules=False):
         if not _repo_has_testing_flag(rdb, omod["repository"]):
             # Ignore testing releases for new installations:
             omod["versions"] = list(filter(lambda v: _tag_is_stable(v["tag"]), omod["versions"]))
-        omod["versions"] = list(filter(lambda v: _min_core_ok(v, leader_core_version), omod["versions"]))
-        if not omod["versions"]:
-            continue # Ignore modules with no versions
+        # Check the min-core label compatibility:
+        omod_core_compat_versions = list(filter(lambda v: _min_core_ok(v, leader_core_version), omod["versions"]))
+        if omod_core_compat_versions:
+            omod["versions"] = omod_core_compat_versions
+        else:
+            # There are no compatible versions because current core is too old
+            try:
+                core_min_required = str(min(list(map(lambda mv: semver.Version.parse(mv['labels']['org.nethserver.min-core']), omod['versions']))))
+            except:
+                core_min_required = "0.0.0"
+            print(agent.SD_WARNING + f"Ignoring image of application {omod['source']}. Installed Core version {leader_core_version} is less than the required version {core_min_required}.", file=sys.stderr)
```

Comment:
Do we still need this print? If we do, I propose to change the message: now the app is not ignored/skipped